examples/js/loaders/GLTFLoader.js

```diff
@@ -2144,442 +2132,409 @@ THREE.GLTFLoader = ( function () {
-		var extensions = this.extensions;
+					mesh.bind( new THREE.Skeleton( bones, boneInverses ), mesh.matrixWorld );
 
-		// scene node hierachy builder
+				}
 
-		function buildNodeHierachy( nodeId, parentObject, allNodes ) {
+			}
 
-			var _node = allNodes[ nodeId ];
-			parentObject.add( _node );
+			// build node hierachy
 
-			var node = json.nodes[ nodeId ];
+			parentObject.add( node );
 
-			if ( node.children ) {
+			if ( nodeDef.children ) {
 
-				var children = node.children;
+				var children = nodeDef.children;
 
-				for ( var i = 0, l = children.length; i < l; i ++ ) {
+				for ( var i = 0, il = children.length; i < il; i ++ ) {
 
 					var child = children[ i ];
-					buildNodeHierachy( child, _node, allNodes );
+					buildNodeHierachy( child, node, json, allNodes, skins );
 
 				}
 
 			}
 
 		}
 
-		return this._withDependencies( [
+		return function loadScene( sceneIndex ) {
 
-			'nodes'
+			var json = this.json;
+			var extensions = this.extensions;
+			var sceneDef = this.json.scenes[ sceneIndex ];
 
-		] ).then( function ( dependencies ) {
+			return this.getMultiDependencies( [
 
-			return _each( json.scenes, function ( scene ) {
+				'node',
+				'skin'
 
-				var _scene = new THREE.Scene();
-				if ( scene.name !== undefined ) _scene.name = scene.name;
+			] ).then( function ( dependencies ) {
 
-				if ( scene.extras ) _scene.userData = scene.extras;
+				var scene = new THREE.Scene();
+				if ( sceneDef.name !== undefined ) scene.name = sceneDef.name;
 
-				var nodes = scene.nodes || [];
+				if ( sceneDef.extras ) scene.userData = sceneDef.extras;
 
-				for ( var i = 0, l = nodes.length; i < l; i ++ ) {
+				var nodeIds = sceneDef.nodes || [];
 
-					var nodeId = nodes[ i ];
-					buildNodeHierachy( nodeId, _scene, dependencies.nodes );
+				for ( var i = 0, il = nodeIds.length; i < il; i ++ ) {
 
-				}
+					buildNodeHierachy( nodeIds[ i ], scene, json, dependencies.nodes, dependencies.skins );
 
-				_scene.traverse( function ( child ) {
-
-					// for Specular-Glossiness.
-					if ( child.material && child.material.isGLTFSpecularGlossinessMaterial ) {
-
-						child.onBeforeRender = extensions[ EXTENSIONS.KHR_MATERIALS_PBR_SPECULAR_GLOSSINESS ].refreshUniforms;
-
-					}
-
-				} );
+				}
 
 				// Ambient lighting, if present, is always attached to the scene root.
-				if ( scene.extensions
-							 && scene.extensions[ EXTENSIONS.KHR_LIGHTS ]
-							 && scene.extensions[ EXTENSIONS.KHR_LIGHTS ].light !== undefined ) {
+				if ( sceneDef.extensions
+						 && sceneDef.extensions[ EXTENSIONS.KHR_LIGHTS ]
+						 && sceneDef.extensions[ EXTENSIONS.KHR_LIGHTS ].light !== undefined ) {
 
 					var lights = extensions[ EXTENSIONS.KHR_LIGHTS ].lights;
-					_scene.add( lights[ scene.extensions[ EXTENSIONS.KHR_LIGHTS ].light ] );
+					scene.add( lights[ sceneDef.extensions[ EXTENSIONS.KHR_LIGHTS ].light ] );
 
 				}
 
-				return _scene;
+				return scene;
 
 			} );
 
-		} );
+		};
 
-	};
+	}();
```

Comment:
This is a pretty long with the IIFE here... maybe `buildNodeHierarchy` should just be a local helper function or a method of `GLTFParser`?

Reply:
For the performance, readability, or anything else?

Reply:
Just readability IMO

Reply:
I didn't really wanna make it `GLTFParser` method because it's used only in `.loadScene()`.
And the reason why I used IIFE is `.loadScene()` can be called many times with glTF including many scenes. I didn't wanna make function for each call.

Reply:
Yeah making a function for each call would be worse I agree. I think the alternative is making a helper function, like `addMorphTargets` that is a local function but not exposed as a method on GLTFParser. Just easier to read with less nesting I think.

Reply:
Ah, OK. I've misunderstood a bit. Yeah, helper function is one of the good ideas. Let's make PR for it if/when it's necessary.